src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/util/StorageStubHttpClient.java

```diff
@@ -14,28 +15,63 @@
     private static final String LIST_CONTAINERS = "?comp=list";
     private static final String LIST_ONE_BLOB = "/bulkscan?include=&restype=container&comp=list";
     private static final String LIST_ZERO_BLOBS = "/empty?include=&restype=container&comp=list";
+    private static final String FILE_1 = "/bulkscan/file1.zip";
+    private static final String FILE_3 = "/bulkscan/file3.zip";
+    private static final String FILE_CAUSING_404 = "/bulkscan/causes_404.zip";
+    private static final String FILE_CAUSING_500 = "/bulkscan/causes_500.zip";
 
     @Override
     public Mono<HttpResponse> send(HttpRequest request) {
+        HttpMethod method = request.getHttpMethod();
         String path = request.getUrl().getPath();
         String query = request.getUrl().getQuery();
 
-        switch (path + "?" + query) {
-            case LIST_CONTAINERS:
-                return Mono.just(
-                    new MockHttpResponse(request, 200, getFileContents("storage/list-containers.json"))
-                );
-            case LIST_ONE_BLOB:
-                return Mono.just(
-                    new MockHttpResponse(request, 200, getFileContents("storage/list-blobs.json"))
-                );
-            case LIST_ZERO_BLOBS:
-                return Mono.just(
-                    new MockHttpResponse(request, 200, getFileContents("storage/list-empty-blobs.json"))
-                );
-            default:
-                Assertions.fail("Request '" + request.getUrl() + "' is not set up");
-                return Mono.empty();
+        if (query == null) {
```

Comment:
Shall we wait for other PRs to get in? all this establishment was under construction and product of all investigations that took course. we can review later